core/arch/arm/pta/benchmark.c

```diff
@@ -57,48 +64,69 @@ static TEE_Result rpc_reg_global_buf(uint64_t type, paddr_t phta, size_t size)
 	return thread_rpc_cmd(OPTEE_MSG_RPC_CMD_BENCH_REG, 1, &rpc_params);
 }
 
-static TEE_Result register_benchmark_memref(uint32_t type,
+static TEE_Result alloc_benchmark_buffer(uint32_t type,
 				TEE_Param p[TEE_NUM_PARAMS])
 {
 	TEE_Result res;
 
-	if ((TEE_PARAM_TYPE_GET(type, 0) != TEE_PARAM_TYPE_MEMREF_INOUT) ||
-		(TEE_PARAM_TYPE_GET(type, 1) != TEE_PARAM_TYPE_NONE) ||
+	if ((TEE_PARAM_TYPE_GET(type, 0) != TEE_PARAM_TYPE_VALUE_INOUT) ||
+		(TEE_PARAM_TYPE_GET(type, 1) != TEE_PARAM_TYPE_VALUE_INPUT) ||
 		(TEE_PARAM_TYPE_GET(type, 2) != TEE_PARAM_TYPE_NONE) ||
 		(TEE_PARAM_TYPE_GET(type, 3) != TEE_PARAM_TYPE_NONE)) {
 		return TEE_ERROR_BAD_PARAMETERS;
 	}
 
-	/*
-	 * We accept only non-secure buffers, as we later perform
-	 * registration of this buffer in NS layers
-	 * (optee linux kmod/optee client)
-	 */
-	if (!tee_vbuf_is_non_sec(p[0].memref.buffer, p[0].memref.size))
-		return TEE_ERROR_BAD_PARAMETERS;
-
 	mutex_lock(&bench_reg_mu);
 
 	/* Check if we have already registered buffer */
 	if (bench_ts_global) {
-		EMSG("Timestamp buffer was already registered\n");
+		EMSG(TA_PRINT_PREFIX
+			"timestamp buffer was already registered\n");
+		mutex_unlock(&bench_reg_mu);
+		return TEE_ERROR_BAD_STATE;
+	}
+
+	bench_ts_size = sizeof(struct tee_ts_global) +
+		(p[1].value.a * sizeof(struct tee_ts_cpu_buf));
+	if (!bench_ts_size) {
+		EMSG(TA_PRINT_PREFIX
+			"timestamp buffer size isn't specified\n");
```

Comment:
The issue looks more like `invalid timestamp buffer size found`.

minor: no need to add ending `\n` in `EMSG()` macro and friends.